acra/security-controls/masking/_index.md

```diff
@@ -2,3 +2,122 @@
+
+# Masking
+
+Partial column encryption.
+Hides encrypted part from clients/AcraServers who don't have right decryption keys.
+In case of decryption failure, encrypted data will be replaced with a placeholder.
+
+Use cases may include:
+* Credit card numbers (reveal only last N digits)
+* Mobile phone (again, reveal only last N digits, i.e. for recovery purposes)
+* First/last name (mask everything except 1-3 first characters)
+
+## AcraServer configuration
+
+In configuration file, passed by `--encryptor_config_file` flag, you can individually configure
+masking for any field of any table, if that field can be encrypted.
+These options are accepted:
+
+<!-- Config struct lives in encryptor/config/encryptionSettings.go -->
+<!-- Config validation func lives in masking/common/patterns.go -->
+```yaml
+{
+    # (required, must be non-empty)
+    # Replacement string to use for masking out encrypted data.
+    # If decryption fails, this string will be show instead of the masked
+    # portion of the data.
+    masking: "XXXXXXXXXXXX",
+
+    # (required, must be non-negative)
+    # How many bytes of plaintext to leave unencrypted.
+    plaintext_length: 4,
+
+    # (required, allowed values: "left", "right")
+    # Which side of the plaintext to leave unencrypted.
+    # "plaintext_length" bytes will be retained from left or right side of
+    # the data cell.
+    plaintext_side: "right",
+
+    # (optional, default: "acrastruct", allowed values: "acrastruct", "acrablock")
+    # Which cryptographic container to use for data encryption.
+    crypto_envelope: "acrastruct",
+
+    # (optional, default: false)
+    # If true, data stored in AcraStructs will be transparently reencrypted
+    # into AcraBlocks.
+    reencrypting_to_acrablocks: false,
+}
+```
+<!-- TODO add link to page where colemn encryption settings are described in general, with client_id, zone_id etc -->
+
+## Examples
+
+Here are few examples of configuration and results it would give.
+The actual encrypted part of data will look differently in database, and will take more bytes.
+Examples below only ilustrate which part of plaintext would be encrypted.
+
+### Retain first five characters
+
+```yaml
+{
+  masking: "*",
+  plaintext_length: 5,
+  plaintext_side: "left"
+}
+```
+
+| Viewpoint                  | Visible data            |
+| -------------------------- | :---------------------- |
+| database, encrypted        | `Alan PGVuY3J5cHRlZD4=` |
+| authorized user, decrypted | `Alan Turing`           |
+| unauthorized used, masked  | `Alan *`                |
+
+### Retain last four characters
+
+```yaml
+{
+  masking: "XXXX",
+  plaintext_length: 4,
+  plaintext_side: "right"
+}
+```
+
+| Viewpoint                  | Visible data            |
+| -------------------------- | :---------------------- |
+| database, encrypted        | `PGVuY3J5cHRlZD4=scal`  |
+| authorized user, decrypted | `Blaise Pascal`         |
+| unauthorized used, masked  | `XXXXscal`              |
```

Comment:
I believe just one example is enough. “Retain last four characters” looks like a perfect fit for masking CC digits, so I'd remove this example with names and leave only the following one.